Use `--gpu-driver none` instead of `--skip-gpu-driver-install` (#54)
Azure/aks-rdma-infiniband: website/docs/configurations/02-gpu-drivers.md

When provisioning GPU nodepools in an AKS cluster, the cluster administrator has the option to either rely on the default GPU driver installation managed by AKS or via GPU Operator. This decision impacts cluster setup, maintenance, and compatibility.

|                | **AKS-managed GPU Driver (Without GPU Operator)**                                  | **GPU Operator-managed GPU Driver (`--gpu-driver none`)**                                 |
|----------------|------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------|
| **Automation** | AKS-managed drivers; cluster administrator needs to manually deploy device plugins | Automates installation of driver, device plugins, and container runtimes via GPU Operator |
| **Complexity** | Simple, no additional components except device plugins                             | More complex, requires GPU Operator and additional components                             |

> | **Support**    | Fully supported by AKS; no preview features                                        | `--gpu-driver none` is a preview feature; limited support available                       |

Reviewer: ```suggestion
| **Support**    | Fully supported by AKS; no preview features                                        | No AKS support; driver install and maintenance with self-managed NVIDIA GPU Operator                    |
```

:::danger

> AKS-managed GPU drivers and the NVIDIA GPU Operator managed GPU drivers are **mutually exclusive** and cannot coexist. When you create a nodepool **without** the `--gpu-driver none` flag, AKS provisions the nodepool with NVIDIA drivers and the NVIDIA container runtime. Installing GPU Operator subsequently replaces this setup by deploying its own `nvidia-container-toolkit`, overriding the AKS-managed configuration. Upon uninstalling GPU Operator, the toolkit cannot revert to the original AKS containerd configuration, as it lacks awareness of the prior state, potentially disrupting the node’s container runtime and impairing workload execution.

Reviewer: ```suggestion
AKS-managed GPU drivers and the NVIDIA GPU Operator managed GPU drivers are **mutually exclusive** and cannot coexist. When you create a nodepool **without** setting the `--gpu-driver` field to `none`, AKS provisions the nodepool with NVIDIA drivers and the NVIDIA container runtime. Installing GPU Operator subsequently replaces this setup by deploying its own `nvidia-container-toolkit`, overriding the AKS-managed configuration. Upon uninstalling GPU Operator, the toolkit cannot revert to the original AKS containerd configuration, as it lacks awareness of the prior state, potentially disrupting the node’s container runtime and impairing workload execution.
```

### Option 1: AKS-managed GPU Driver

:::warning

> Please proceed with GPU operator installation only if you have created the nodepool **without** the `--gpu-driver none` flag as described in [prerequisites documentation](../getting-started/02-prerequisites.md#aks-managed-gpu-driver).

Reviewer: ```suggestion
Please proceed with GPU operator installation only if you have set the `--gpu-driver` field to `none` as described in [prerequisites documentation](../getting-started/02-prerequisites.md#aks-managed-gpu-driver).
```

:::

:::warning

> Please proceed with GPU operator installation only if you have created the nodepool **with** the `--gpu-driver none` flag as described in [prerequisites documentation](../getting-started/02-prerequisites.md#gpu-operator-managed-gpu-driver).

Reviewer: ```suggestion
Please proceed with GPU operator installation only if you have created the nodepool **with** the `--gpu-driver` field set to `none` as described in [prerequisites documentation](../getting-started/02-prerequisites.md#gpu-operator-managed-gpu-driver).
```
